homeassistant/components/lovelace/__init__.py

```diff
@@ -1,19 +1,95 @@
 """Lovelace UI."""
+import logging
+import uuid
+import os
+from os import O_WRONLY, O_CREAT, O_TRUNC
+from collections import OrderedDict
+from typing import Union, List, Dict
 import voluptuous as vol
 
 from homeassistant.components import websocket_api
-from homeassistant.util.yaml import load_yaml
 from homeassistant.exceptions import HomeAssistantError
 
+_LOGGER = logging.getLogger(__name__)
 DOMAIN = 'lovelace'
+REQUIREMENTS = ['ruamel.yaml==0.15.72']
 
 OLD_WS_TYPE_GET_LOVELACE_UI = 'frontend/lovelace_config'
 WS_TYPE_GET_LOVELACE_UI = 'lovelace/config'
+
 SCHEMA_GET_LOVELACE_UI = websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
     vol.Required('type'): vol.Any(WS_TYPE_GET_LOVELACE_UI,
                                   OLD_WS_TYPE_GET_LOVELACE_UI),
 })
 
+JSON_TYPE = Union[List, Dict, str]  # pylint: disable=invalid-name
+
+
+class WriteError(HomeAssistantError):
+    """Error writing the data."""
+
+
+def save_yaml(fname: str, data: JSON_TYPE):
```

Comment:
So there is a lot of copy paste going on in these methods. I think that it's fine for now.

A bunch of that will go away once we migrate the util.yaml to `ruamel.yaml`, as it implements YAML 1.2 and not YAML 1.1.

Reply:
btw, I can only recommend `ruamel.yaml`. Recently started porting esphomeyaml to use it and it's rather simple - since esphomeyaml uses almost the same loader/yaml_util code (😄), I can then also port it to home assistant

Reply:
That would be awesome Otto.